src/hooks/cakeVault/useWithdrawalFeeTimer.ts

```diff
@@ -3,17 +3,25 @@ import { useEffect, useState } from 'react'
 const useWithdrawalFeeTimer = (lastDepositedTime: number, withdrawalFeePeriod = 259200) => {
   const [secondsRemaining, setSecondsRemaining] = useState(null)
   const [hasUnstakingFee, setHasUnstakingFee] = useState(false)
+  const [currentSeconds, setCurrentSeconds] = useState(Math.floor(Date.now() / 1000))
 
   useEffect(() => {
-    const threeDaysFromDeposit = lastDepositedTime + withdrawalFeePeriod
-    const now = Math.floor(Date.now() / 1000)
-    const secondsRemainingCalc = threeDaysFromDeposit - now
+    const tick = () => {
+      setCurrentSeconds((prevSeconds) => prevSeconds + 1)
+    }
+    const timerInterval = setInterval(() => tick(), 1000)
+    return () => clearInterval(timerInterval)
+  }, [])
+
+  useEffect(() => {
+    const feeEndTime = lastDepositedTime + withdrawalFeePeriod
+    const secondsRemainingCalc = feeEndTime - currentSeconds
     const doesUnstakingFeeApply = secondsRemainingCalc > 0
     if (doesUnstakingFeeApply) {
       setSecondsRemaining(secondsRemainingCalc)
       setHasUnstakingFee(true)
     }
-  }, [lastDepositedTime, withdrawalFeePeriod, setSecondsRemaining])
```

Comment:
I don't see a clear reason why this needs to be 2 hooks. Am I missing something?

Reply:
Yea I thought that setting the interval within the same `useEffect` that has the `currentSeconds` dependency would create new intervals every time `currentSeconds` is updated - creating a rerender loop..

But it looks like the `return` also cleans up effects from the previous render before running the effects next time? Interesting, I only thought it did this on unmount. 

Is it not less performance to have this all within the one effect, if it's having to clear & set an interval each time the effect fires (happening every 1000ms), rather than just once on mount? I've pushed up this change regardless.